sycl/include/sycl/multi_ptr.hpp

```diff
@@ -1546,70 +1546,80 @@ template <typename ElementType, access::address_space Space,
           access::decorated DecorateAddress>
 bool operator==(const multi_ptr<ElementType, Space, DecorateAddress> &lhs,
                 std::nullptr_t) {
-  return lhs.get() == nullptr;
+  return lhs.get() ==
+         multi_ptr<ElementType, Space, DecorateAddress>(nullptr).get();
```

Comment:
I don't understand that change here. With sycl tip, the following

```cpp
__attribute__((sycl_device)) bool f(__attribute__((opencl_local))int* i) {
    return i == nullptr;
}
```

generates this for nvptx

```llvm
%cmp = icmp eq ptr addrspace(3) %i, addrspacecast (ptr null to ptr addrspace(3))
```

which is what we expect. If you have an issue `lhs.get() == nullptr`, this probably hides a compiler issue we need logged.

Reply:
thanks @Naghasan 
You are right. There was no need for this change for `==`. For other operators having something like `lhs.get() > nullptr` results in build error.

Reply:
Thanks @Naghasan 
You are right  regarding `==` operator. It fails to build for other relational operators such as `lhs.get() > nullptr`.

Reply:
thanks @Naghasan , that particular change is reverted.